validator_client/src/notifier.rs

```diff
@@ -11,86 +11,88 @@ pub fn spawn_notifier<T: EthSpec>(client: &ProductionValidatorClient<T>) -> Resu
     let duties_service = client.duties_service.clone();
 
     let slot_duration = Duration::from_secs(context.eth2_config.spec.seconds_per_slot);
-    let duration_to_next_slot = duties_service
-        .slot_clock
-        .duration_to_next_slot()
-        .ok_or("slot_notifier unable to determine time to next slot")?;
-
-    // Run the notifier half way through each slot.
-    let start_instant = Instant::now() + duration_to_next_slot + (slot_duration / 2);
-    let mut interval = interval_at(start_instant, slot_duration);
 
     let interval_fut = async move {
         let log = context.log();
 
         loop {
-            interval.tick().await;
-            let num_available = duties_service.beacon_nodes.num_available().await;
-            let num_synced = duties_service.beacon_nodes.num_synced().await;
-            let num_total = duties_service.beacon_nodes.num_total().await;
-            if num_synced > 0 {
-                info!(
-                    log,
-                    "Connected to beacon node(s)";
-                    "total" => num_total,
-                    "available" => num_available,
-                    "synced" => num_synced,
-                )
-            } else {
-                error!(
-                    log,
-                    "No synced beacon nodes";
-                    "total" => num_total,
-                    "available" => num_available,
-                    "synced" => num_synced,
-                )
-            }
+            match duties_service.slot_clock.duration_to_next_slot() {
```

Comment:
We could fight a bit of rightward drift by using an `if let` here (it's already pretty nested). But happy to merge either way.